iphone/Maps/Classes/ClipboardMonitor.mm

```diff
@@ -0,0 +1,107 @@
+#import "MapsAppDelegate.h"
+#import "ClipboardMonitor.h"
+#import "UIKitCategories.h"
+#import "MapsAppDelegate.h"
+#import <CoreApi/DeepLinkParser.h>
+#include <CoreApi/Framework.h>
+#import <CoreApi/DeepLinkData.h>
+#import "UIKitCategories.h"
+#import <CoreApi/DeepLinkSearchData.h>
+
+@interface ClipboardWatcher ()
+
+@property (nonatomic, strong) NSTimer *timer;
+@property (nonatomic, copy) NSString *previousClipboardText;
+@property (nonatomic, assign) BOOL isErrorPromptShown;
+
+@end
+
+@implementation ClipboardWatcher
+
+- (void)startWatching:(UIWindow *)window {
+    NSLog(@"clip watching started");
+
+    UIPasteboard *pasteboard = [UIPasteboard generalPasteboard];
+    NSString *currentClipboardText = pasteboard.string;
+    NSLog(@"clip clipboard checking for changes %@  and  %@",currentClipboardText, self.previousClipboardText);
+
+    [pasteboard setString:@""];
+    [self checkClipboardWithText:currentClipboardText window:window];
+}
+
+- (void)checkClipboardWithText:(NSString *)copiedText window:(UIWindow *)window {
+    NSLog(@"clip API CALL PROCEEDING %@", copiedText);
+
+    // Check if the copied text is a URL
+    if ([self isURL:copiedText]) {
+
+        dispatch_async(dispatch_get_main_queue(), ^{
+            UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Redirecting from copied URL" message:@"Please wait..." preferredStyle:UIAlertControllerStyleAlert];
+            [window.rootViewController presentViewController:alertController animated:YES completion:nil];
+        });
+
+        // Make an API request
+        NSURL *apiURL = [NSURL URLWithString:[NSString stringWithFormat:@"https://url-un.kartikay-2101ce32.workers.dev/search?url=%@", copiedText]];
+        NSURLSessionDataTask *task = [[NSURLSession sharedSession] dataTaskWithURL:apiURL completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+            dispatch_async(dispatch_get_main_queue(), ^{
+                [window.rootViewController dismissViewControllerAnimated:YES completion:nil];
+            });
+
+            if (error) {
+                NSLog(@"API Request Error: %@", error);
+                NSString *errorMessage = @"Kindly make sure that your internet connection in turned on.";
```

Comment:
"Failed to extract coordinates from %url using an anonymous proxy. Please check your Internet connection"